packages/webapp/src/containers/Task/saga.js

```diff
@@ -364,13 +364,84 @@ const getTransplantTaskBody = (data, endpoint, managementPlanWithCurrentLocation
   );
 };
 
+const getIrrigationTaskBody = (data, endpoint, managementPlanWithCurrentLocationEntities) => {
+  const managementPlans = data.managementPlans.map(
+    (managementPlan) => managementPlan.management_plan_id,
+  );
+  const irrigation_task_type =
+    data.irrigation_task_type.value !== 'OTHER'
+      ? data.irrigation_task_type.value
+      : data.irrigation_task_type_other;
+  const set_default_irrigation_task_location = data.set_default_irrigation_task_type_location;
+  const set_default_irrigation_task_type_measurement =
+    data.set_default_irrigation_task_type_measurement;
+  const default_flow_rate = data.default_location_flow_rate;
+  const flow_rate = data.estimated_flow_rate;
+  const flow_rate_unit = data.estimated_flow_rate_unit;
+  const default_application_depth = data.default_location_application_depth;
+  const application_depth = data.application_depth;
+  const application_depth_unit = data.application_depth_unit;
+  return produce(
+    getPostTaskBody(data, 'irrigation_task', managementPlanWithCurrentLocationEntities),
+    (data) => {
+      data.managementPlans = managementPlans?.map((management_plan_id) => ({
+        planting_management_plan_id:
+          managementPlanWithCurrentLocationEntities[management_plan_id].planting_management_plan
+            .planting_management_plan_id,
+      }));
+      data.irrigation_task = {
+        type: irrigation_task_type,
+        default_measuring_type: data.measurement_type,
+      };
+      data.irrigation_type = {
+        irrigation_type_name: irrigation_task_type,
+      };
+      set_default_irrigation_task_type_measurement &&
+        (data.irrigation_task.irrigation_type = data.measurement_type);
+
+      data.location = {
+        location_id: data.locations[0].location_id,
+        flow_rate,
+        flow_rate_unit,
+        application_depth,
+        application_depth_unit,
+        default_application_depth,
+        default_flow_rate,
+      };
+      set_default_irrigation_task_location &&
+        (data.location.irrigation_task_type = irrigation_task_type);
+      delete data.irrigation_task_type_other;
```

Comment:
can take out the useful values in a separate object and override it with the data.